engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/typed/AvroSchemaTypeDefinitionExtractor.scala

```diff
@@ -20,26 +20,28 @@ class AvroSchemaTypeDefinitionExtractor(skippNullableFields: Boolean) {
   import collection.JavaConverters._
 
   // see BestEffortAvroEncoder for underlying avro types
-  def typeDefinition(schema: Schema): TypingResult = {
+  def typeDefinition(schema: Schema, possibleTypes: Set[TypedClass]): TypingResult = {
     schema.getType match {
       case Schema.Type.RECORD => {
-        val fields = schema.getFields.asScala.filterNot(field => skippNullableFields && field.schema().isNullable)
-        TypedObjectTypingResult(
-          fields.map { field =>
-            field.name() -> typeDefinition(field.schema())
-          }.toMap,
-          Typed.typedClass[GenericRecord]
-        )
+        val fields = schema
+          .getFields
+          .asScala
+          //@TODO: Maybe Optional field should be marked when it has defaultValue?
+          .filterNot(field => skippNullableFields && field.schema().isNullable && field.hasDefaultValue)
```

Comment:
why only skip when both default value and isNullable?

Reply:
When field is `union: [null, string]` you still should set value ;) So it's not definition of optional field..

Reply:
For example PaymentV1.schema has field vat. When we remove setting null then avro throws exception:
```
Field vat type:UNION pos:5 not set and has no default value
Path in schema: --> vat